.github/workflows/pull_request.yml

```diff
@@ -169,6 +162,14 @@ jobs:
           cd "$GITHUB_WORKSPACE/tests/ci"
           python3 merge_pr.py --set-ci-status --wf-status ${{ contains(needs.*.result, 'failure') && 'failure' || 'success' }}
 
+  Tests_3:
+    needs: [RunConfig, Builds_1, CheckReadyForMerge]
```

Comment:
Why did we change dep to `CheckReadyForMerge`?

Reply:
In #66440 I don't want `Tests_3` be blocked by `Tests_1` if a small number of tests fail. The number of failed tests is checked in the `CheckReadyForMerge` job.
But this change doesn't work yet because `Tests_3` can still be blocked by a failed dependency "Tests_1", no matter how many tests fail.
So in this PR I want to remove the direct dependency on `Tests_1` and run `Tests_3` if `CheckReadyForMerge` is green. I'm not sure yet if will work out, I need to test it